streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionWrapperSerde.java

```diff
@@ -63,9 +63,9 @@ public SubscriptionWrapperSerde(final Supplier<String> primaryKeySerializationPs
         }
 
         @Override
-        public void setIfUnset(final Serializer<K> defaultSerializer) {
+        public void setIfUnset(final Serializer<K> defaultKeySerializer, final Serializer<Void> defaultValueSerializer) {
             if (primaryKeySerializer == null) {
-                primaryKeySerializer = Objects.requireNonNull(defaultSerializer, "defaultSerializer cannot be null");
+                primaryKeySerializer = Objects.requireNonNull(defaultKeySerializer);
```

Comment:
@guozhangwang , this (and L128) is the "actual fix". Previously, the actual default serde we would pass in was the value serde (in SinkNode/SourceNode), but this one particular serde needs the key serde. If you look at the other serdes that changed in this diff, you'll note that they all actually do need the value serde. Only this one is different.

Reply:
Oh, hah. Looks like you beat me to it, @bellemare  :)